saucerswap-faucetbot/main.js

```diff
@@ -0,0 +1,373 @@
+          userMsg
+        );
+        break;
+      }
+
+      // regex check for accountId
+      if (treasuryId.toString() == accountId) {
+        replyThenDelete(`I don't need no dang spare change!`, userMsg);
+        break;
+      }
+
+      // Query faucet_tracker table to see if accountId already pulled from the faucet
+      const result = await db.query(
+        `SELECT * FROM charity_tracker WHERE accountId=?`,
+        [accountId]
+      );
+
+      if (result[0].length) {
+        replyThenDelete(
+          `Your account already received its allotment of HBARs - try swapping for more.`,
+          userMsg
+        );
+        break;
+      }
+
+      // check if accountId is currently being processed
+      if (accountIdsBeingProcessed.has(accountId)) {
+        replyThenDelete(`Quit your damn spamming! D:<`, userMsg);
+        break;
+      }
+
+      // add account to set of accountIds being processed
+      // remove accountId before every break statement
+      accountIdsBeingProcessed.add(accountId);
+
+      const data = await fetch(
+        `https://api.coingecko.com/api/v3/simple/price?ids=hedera-hashgraph&vs_currencies=usd`
+      ).then((res) => res.json());
+      const { usd: hbarPrice } = data["hedera-hashgraph"];
+      const hbarPayout = (0.12 / hbarPrice).toFixed(4);
+      if (!hbarPrice || !hbarPayout) {
+        replyThenDelete(`Something funny's going on with the HBAR price...! Try again?`, userMsg);
+        accountIdsBeingProcessed.delete(accountId);
+      }
+
+      try {
+        const tx = new TransferTransaction()
+          .addHbarTransfer(operatorId, new Hbar(-hbarPayout))
+          .addHbarTransfer(accountId, new Hbar(hbarPayout));
+        const txResponse = await tx.execute(client);
+        const receipt = await txResponse.getReceipt(client);
+        // if tx success, insert values into table...
+        if (receipt.status._code == 22) {
+          await db.query(`INSERT INTO charity_tracker (accountId) VALUE (?);`, [accountId]);
+          console.debug(
+            `${hbarPayout} HBARs sent to ${accountId}`
+          );
+          replyThenDelete("**Success!** Use your new HBARs:\n"
+          +"\n1. Associate WHBAR (TokenID 0.0.1062664)"
+          +"\n2. Navigate to <https://saucerswap.finance> and swap your existing tokens for HBAR (this will use the WHBAR associated in Step 1)"
+          +"\n3. Use the HBAR you got from Step 2 to fuel your journey on SaucerSwap!\n"
+          +"\nThe HBAR you received will pay for the association fee, network fee, and gas fee incurred by the process outlined above.\n"
+          +"\nWelcome to SaucerSwap! <:SAUCE:946907336742682624>", userMsg);
+          accountIdsBeingProcessed.delete(accountId);
+        } else {
+          throw new Error(`Tx receipt status is not a success!`);
+        }
+      } catch (e) {
+        replyThenDelete(
+          `HBARs couldn't be dispensed! <@453749295309389834> HEEEEELP!!! <:aaa:921863627655508008>`,
```

Comment:
make this a conifg

Reply:
![image](https://user-images.githubusercontent.com/12617332/197709954-e19d9706-6e65-4fa4-b7f1-7af8f86fbd37.png)

That last part of the string is used to alert a developer via discord if the bot fails for whatever reason. I'm removing it if other devs want to customize it to their liking.